staging/src/k8s.io/client-go/rest/request_test.go

```diff
@@ -341,20 +341,51 @@ func TestResultIntoWithNoBodyReturnsErr(t *testing.T) {
 
 func TestURLTemplate(t *testing.T) {
 	uri, _ := url.Parse("http://localhost")
-	r := NewRequest(nil, "POST", uri, "", ContentConfig{GroupVersion: &schema.GroupVersion{Group: "test"}}, Serializers{}, nil, nil, 0)
-	r.Prefix("pre1").Resource("r1").Namespace("ns").Name("nm").Param("p0", "v0")
-	full := r.URL()
-	if full.String() != "http://localhost/pre1/namespaces/ns/r1/nm?p0=v0" {
-		t.Errorf("unexpected initial URL: %s", full)
-	}
-	actualURL := r.finalURLTemplate()
-	actual := actualURL.String()
-	expected := "http://localhost/pre1/namespaces/%7Bnamespace%7D/r1/%7Bname%7D?p0=%7Bvalue%7D"
-	if actual != expected {
-		t.Errorf("unexpected URL template: %s %s", actual, expected)
-	}
-	if r.URL().String() != full.String() {
-		t.Errorf("creating URL template changed request: %s -> %s", full.String(), r.URL().String())
+	testCases := []struct {
+		Request          *Request
+		ExpectedFullURL  string
+		ExpectedFinalURL string
+	}{
+
+		{
+			Request: NewRequest(nil, "POST", uri, "", ContentConfig{GroupVersion: &schema.GroupVersion{Group: "test"}}, Serializers{}, nil, nil, 0).
+				Prefix("pre1").Resource("r1").Namespace("ns").Name("nm").Param("p0", "v0"),
+			ExpectedFullURL:  "http://localhost/pre1/namespaces/ns/r1/nm?p0=v0",
+			ExpectedFinalURL: "http://localhost/pre1/namespaces/%7Bnamespace%7D/r1/%7Bname%7D?p0=%7Bvalue%7D",
```

Comment:
pre1 doesn't match either form of the api paths, so I'd expect this not to get replaced?

Reply:
This was in the test case before and it doesn't matter at the time since it's not a dynamic client, everything is in the struct, so either it's a valid form or not doesn't matter. But it would definitely be better to use a valid form for less confusion. I will change it.

Reply:
I'm actually claiming that it's good to test this string, but the detection logic should not match it, because it isn't an api path.

Reply:
For NON dynamic resource client, the detection logic is to check r.namespaceSet and length of r.namespace, 
`if r.namespaceSet && len(r.namespace) != 0 {
		r.namespace = "{namespace}"
	} `

it did not check if the request is actually valid. I assume it would be checked somewhere else. 

In the case of dynamic resource client, I have a check for that and I will add a test case that doesn't match either legacy or named group.